app/src/main/java/com/example/habitsmasher/LoginFragment.java

```diff
@@ -74,15 +87,150 @@ public View onCreateView(@NonNull LayoutInflater inflater,
+                 @Override
+                 public void onComplete(@NonNull Task<AuthResult> task) {
+                     if (task.isSuccessful()) {
+                         // if log in successful, redirect to home page
+                         showMessage(LOGIN_SUCCESSFUL_MESSAGE, Toast.LENGTH_SHORT);
+
+                         DocumentReference userRef = getCurrentUserDocumentReference();
+                         userRef.get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
+                             @Override
+                             public void onSuccess(DocumentSnapshot documentSnapshot) {
+                                 GoogleSignInAccount user = GoogleSignIn.getLastSignedInAccount(
+                                         getContext());
+
+                                 User googleUser = new User(user.getId(),
+                                                            user.getDisplayName()
+                                                                .replaceAll("\\s+",""),
+                                                            user.getEmail(),
+                                                            GOOGLE_USER_PWD);
+
+                                 // once the user is successfully retrieved, add them to the db
+                                 addUserToDatabaseIfUserDoesNotExist(googleUser);
+
+                                 // save their information to be used globally throughout the app
+                                 saveUserInformation(googleUser);
+
+                                 // transition to the home screen
+                                 navigateToFragmentWithAction(R.id.action_navigation_login_to_HomeFragment);
+                             }
+                         });
+                     } else {
+                         showMessage(INCORRECT_EMAIL_PASSWORD_MESSAGE, Toast.LENGTH_LONG);
+                     }
+                     _progressBar.setVisibility(View.GONE);
+                     _bottomNav.setVisibility(View.VISIBLE);
+                 }
+             });
+    }
+
+    /**
+     * This helper method adds the google user to the database if they don't already exist
+     * @param googleUser the user to add
+     */
+    private void addUserToDatabaseIfUserDoesNotExist(User googleUser) {
+        FirebaseFirestore.getInstance()
+                         .collection("Users")
+                         .document(googleUser.getId()).get().addOnCompleteListener(
+                new OnCompleteListener<DocumentSnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                        if (!task.isSuccessful()) {
+                            // if the user does not exist already, add them to the db collection
+                            _userAccountHelper.addNewUserToDatabase(googleUser);
+                        }
```

Comment:
We only want to add the user to the user collection in the db if they haven't been added already!